src/main/java/com/minecolonies/util/StructureWrapper.java

```diff
@@ -275,141 +235,38 @@ private void placeBlock(IBlockState localState, @NotNull Block localBlock, @NotN
-        int minY = Math.min(from.getY(), to.getY());
-        int maxY = Math.max(from.getY(), to.getY());
-        int minZ = Math.min(from.getZ(), to.getZ());
-        int maxZ = Math.max(from.getZ(), to.getZ());
-        short width = (short) (Math.abs(maxX - minX) + 1);
-        short height = (short) (Math.abs(maxY - minY) + 1);
-        short length = (short) (Math.abs(maxZ - minZ) + 1);
-
-        @NotNull BlockPos minPos = new BlockPos(minX, minY, minZ);
-
-        @NotNull Schematic schematic = new Schematic(DEFUALT_ICON, width, height, length);
-
-        @NotNull BlockPos.MutableBlockPos offset = new BlockPos.MutableBlockPos(0, 0, 0);
-
-        for (int x = minX; x <= maxX; x++)
-        {
-            for (int y = minY; y <= maxY; y++)
-            {
-                for (int z = minZ; z <= maxZ; z++)
-                {
-                    @NotNull BlockPos worldPos = new BlockPos(x, y, z);
-                    BlockPos localPos = worldPos.subtract(minPos);
-
-                    IBlockState blockState = world.getBlockState(worldPos);
-                    schematic.setBlockState(localPos, blockState);
-
-                    if (blockState.getBlock() instanceof AbstractBlockHut)
-                    {
-                        if (BlockPosUtil.isEqual(offset, 0, 0, 0))
-                        {
-                            BlockPosUtil.set(offset, localPos);
-                        }
-                        else
-                        {
-                            schematic.setBlockState(localPos, Blocks.air.getDefaultState());
-                            Log.getLogger().warn("Scan contained multiple AbstractBlockHut's ignoring this one");
-                        }
-                    }
-
-                    TileEntity tileEntity = world.getTileEntity(worldPos);
-                    saveTileEntity(schematic, tileEntity, minPos);
-                }
-            }
-        }
-        if (BlockPosUtil.isEqual(offset, 0, 0, 0))
-        {
-            offset.set(width / TWO_FOR_HALVING, 0, length / TWO_FOR_HALVING);
-        }
-        schematic.setOffset(offset);
-
-        @NotNull AxisAlignedBB region = AxisAlignedBB.fromBounds(minX, minY, minZ, maxX, maxY, maxZ);
-        //schematic::addEntity already does null checking and ignores null values.
-        world.getEntitiesWithinAABB(EntityHanging.class, region).forEach(schematic::addEntity);
-        world.getEntitiesWithinAABB(EntityMinecart.class, region).forEach(schematic::addEntity);
-
-        return schematic;
-    }
-
-    @NotNull
-    private static File getScanDirectory(@NotNull World world)
-    {
-        File minecolonies;
-        if (world.isRemote)
-        {
-            minecolonies = new File(Minecraft.getMinecraft().mcDataDir, "minecolonies/");
-        }
-        else
-        {
-            minecolonies = MinecraftServer.getServer().getFile("minecolonies/");
-        }
-        checkDirectory(minecolonies);
+        BlockPos blockpos =
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'blockpos1' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)